packages/presentation/src/RevisionSwitcher.tsx

```diff
@@ -24,20 +24,16 @@ export const RevisionSwitcher: FunctionComponent<RevisionSwitcherProps> = functi
 
   const editState = useEditState(documentId, documentType)
 
-  useEffectOnChange(
-    perspective,
-    (value) => {
-      let rev: string | undefined = undefined
-      if (value === 'published' && editState.published) {
-        const {_updatedAt, _rev} = editState.published
-        rev = `${_updatedAt}/${_rev}`
-      }
-      if (documentRevision !== rev) {
-        navigate({}, {rev}, true)
-      }
-    },
-    [editState, navigate, documentRevision],
```

Comment:
The React Compiler doesn't understand custom hooks that roll their own `React.DependencyList` related logic.
In this case I was able to change `useEffectOnChange` internally to no longer need memoization.
But if that weren't an option, the solution would've been to lift up `useCallback`:
```ts
useEffectOnChange(
  perspective,
  useCallback(
    (value) => {
      let rev: string | undefined = undefined
      if (value === 'published' && editState.published) {
        const {_updatedAt, _rev} = editState.published
        rev = `${_updatedAt}/${_rev}`
      }
      if (documentRevision !== rev) {
        navigate({}, {rev}, true)
      }
    },
    [editState, navigate, documentRevision]
  )
)
```